lib/facility.rb

```diff
@@ -1,14 +1,84 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
-
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  attr_reader :name, :address, :phone
+  attr_accessor :services, :registered_vehicles
+  def initialize(args)
+    args.each do |k, v|
+    instance_variable_set("@#{k}", v) unless v.nil?
+    end
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def register_vehicle(vehicle)
+    @registered_vehicles << vehicle
+    vehicle.registration_date = Date.today.year
+    if @services.include?('Vehicle Registration') == false
+      return nil
+    elsif vehicle.antique? == true
+      @collected_fees += 25
+      vehicle.plate_type = :antique
+     elsif
+       vehicle.electric_vehicle? == true
+       @collected_fees += 200
+       vehicle.plate_type = :ev
+    else
+      @collected_fees += 100
+      vehicle.plate_type = :regular
+    end
+  end
```

Comment:
Good work. This could also be a fun place to implement helper methods. Maybe something like: 

```ruby
    if @services.include?('Vehicle Registration') == false
      return nil
    elsif vehicle.antique? == true
     antique_registration(vehicle)
     elsif
       vehicle.electric_vehicle? == true
       ev_registration(vehicle)
    else
      reg_registration(vehicle)
    end
```